src/main/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -41,10 +44,19 @@ import java.util.Optional
 
 internal class YamlOutput(
     writer: StreamDataWriter,
-    private val configuration: YamlConfiguration
+    private val yaml: Yaml
```

Comment:
`YamlInput` takes the configuration and context as properties, rather than receiving the whole `Yaml` instance - it would be good to keep it consistent with that